source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -472,8 +474,12 @@ ShaderGraphPtr ShaderGraph::createSurfaceShader(
     // Create this shader node in the graph.
     const string& newNodeName = node->getName();
     ShaderNodePtr newNode = ShaderNode::create(graph.get(), newNodeName, *nodeDef, context);
+    newNode->initialize(*node, *nodeDef, context);
     graph->addNode(newNode);
 
+    // Share metadata.
+    graph->setMetadata(newNode->getMetadata());
```

Comment:
I'm not sure I understand what it means to assign the node meta data to the graph as it will keep on changing for each new node added won't it?

Reply:
In this case we are creating a surface shader, so newNode here is an instance of the nodedef for the surface shader. And when created the newNode has metadata setup from the attributes declared on the nodedef. However in order for this to reach the graph interface (which ultimately is the interface that gets generated into inputs on the shader) the graph gets a copy of the same metadata container here.

Reply:
So this happens only once, for the single node that represents the surface shader, and not for every other node in the graph.